### PR TITLE
HHH-16515 - More nullability work

### PR DESCRIPTION
**Pull request by jrenaat**
Add o.h.property.access to nullness checking (internal + spi)

https://hibernate.atlassian.net/browse/HHH-16515

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/property/access/internal/PropertyAccessMixedImpl.java [hunk 1]
```diff
@@ -74,19 +76,19 @@ public PropertyAccessMixedImpl(PropertyAccessStrategy strategy, Class<?> contain
 
 	// --- //
 
-	protected Getter fieldGetter(Class<?> containerJavaType, String propertyName, Field field) {
+	protected Getter fieldGetter(@UnderInitialization PropertyAccessMixedImpl this, Class<?> containerJavaType, String propertyName, Field field) {
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/property/access/internal/PropertyAccessMixedImpl.java:79]:
```suggestion
	private static Getter fieldGetter(Class<?> containerJavaType, String propertyName, Field field) {
```

File: hibernate-core/src/main/java/org/hibernate/property/access/internal/PropertyAccessMixedImpl.java [hunk 2]
```diff
@@ -74,19 +76,19 @@ public PropertyAccessMixedImpl(PropertyAccessStrategy strategy, Class<?> contain
...
 		return new GetterFieldImpl( containerJavaType, propertyName, field );
 	}
 
-	protected Setter fieldSetter(Class<?> containerJavaType, String propertyName, Field field) {
+	protected Setter fieldSetter(@UnderInitialization PropertyAccessMixedImpl this, Class<?> containerJavaType, String propertyName, Field field) {
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/property/access/internal/PropertyAccessMixedImpl.java:83]:
```suggestion
	private static Setter fieldSetter(Class<?> containerJavaType, String propertyName, Field field) {
```

File: hibernate-core/src/main/java/org/hibernate/property/access/internal/PropertyAccessMixedImpl.java [hunk 3]
```diff
@@ -74,19 +76,19 @@ public PropertyAccessMixedImpl(PropertyAccessStrategy strategy, Class<?> contain
...
 		return new SetterFieldImpl( containerJavaType, propertyName, field );
 	}
 
-	protected Getter propertyGetter(Class<?> containerJavaType, String propertyName, Method method) {
+	protected Getter propertyGetter(@UnderInitialization PropertyAccessMixedImpl this, Class<?> containerJavaType, String propertyName, Method method) {
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/property/access/internal/PropertyAccessMixedImpl.java:87]:
```suggestion
	private static Getter propertyGetter(Class<?> containerJavaType, String propertyName, Method method) {
```

File: hibernate-core/src/main/java/org/hibernate/property/access/internal/PropertyAccessMixedImpl.java [hunk 4]
```diff
@@ -74,19 +76,19 @@ public PropertyAccessMixedImpl(PropertyAccessStrategy strategy, Class<?> contain
...
 		return new GetterMethodImpl( containerJavaType, propertyName, method );
 	}
 
-	protected Setter propertySetter(Class<?> containerJavaType, String propertyName, Method method) {
+	protected @PolyNull Setter propertySetter(@UnderInitialization PropertyAccessMixedImpl this, Class<?> containerJavaType, String propertyName, @PolyNull Method method) {
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/property/access/internal/PropertyAccessMixedImpl.java:91]:
```suggestion
	private static @PolyNull Setter propertySetter(Class<?> containerJavaType, String propertyName, @PolyNull Method method) {
```

File: hibernate-core/src/main/java/org/hibernate/property/access/internal/ChainedPropertyAccessImpl.java [hunk 5]
```diff
@@ -46,7 +49,11 @@ public Setter getSetter() {
...
 	public Object get(Object owner) {
 		for ( int i = 0; i < propertyAccesses.length; i++ ) {
-			owner = propertyAccesses[i].getGetter().get( owner );
+			// NOTE: I'm not 100% sure the cast to NonNull is correct, if incorrect then we'd have to annotate the method param
+			// with @Nullable, which requires to also annotate the interface method with that, and hence all that hangs from it
+			// This is a bit bizarre since the get method here is a NonNull covariant override of the interface method, but i guess
+			// CF only looks at the latter in this case?
+			owner = NullnessUtil.castNonNull( propertyAccesses[i].getGetter().get( owner ) );
 		}
 		return owner;
```
[the remark this answers is not in the document]
Comment (jrenaat) [reply hibernate-core/src/main/java/org/hibernate/property/access/internal/ChainedPropertyAccessImpl.java:58]: That doesn't work

**Review by beikov — APPROVED** [review]

File: hibernate-core/src/main/java/org/hibernate/property/access/internal/PropertyAccessStrategyResolverStandardImpl.java [hunk 6]
```diff
@@ -79,7 +80,7 @@ protected StrategySelector strategySelectorService() {
 			if ( serviceRegistry == null ) {
 				throw new HibernateException( "ServiceRegistry not yet injected; PropertyAccessStrategyResolver not ready for use." );
 			}
-			strategySelectorService = serviceRegistry.getService( StrategySelector.class );
+			strategySelectorService = NullnessUtil.castNonNull( serviceRegistry.getService( StrategySelector.class ) );
```
Comment (jrenaat) [inline hibernate-core/src/main/java/org/hibernate/property/access/internal/PropertyAccessStrategyResolverStandardImpl.java:83]: @beikov this is a new change that came in when rebaseing the previous nullability work.
Correct if wrong please

**Review by beikov — APPROVED** [review]

**Pull request merged**